src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -598,23 +607,37 @@ bool isSupportedConversion(Conversion conversion, BoundExpression operandOpt)
                     case ConversionKind.Identity:
                     case ConversionKind.DefaultOrNullLiteral:
                     case ConversionKind.ImplicitReference:
+                    case ConversionKind.ExplicitReference:
                         return true;
                     case ConversionKind.ImplicitTupleLiteral:
+                    case ConversionKind.ExplicitTupleLiteral:
+                        switch (operandOpt.Kind)
```

Comment:
Given the perf testing that Neal and Andy have done, and that we're just immediately casting these to the result type, consider making this a type test switch instead.

Reply:
Will leave as is to match outer `switch`.

---
In reply to: [264880390](https://github.com/dotnet/roslyn/pull/34029#discussion_r264880390) [](ancestors = 264880390)